pkgs/development/tools/build-managers/bazel/default.nix

```diff
@@ -186,6 +186,10 @@ stdenv.mkDerivation rec {
   checkPhase = ''
     export TEST_TMPDIR=$(pwd)
     ./output/bazel test --test_output=errors \
+        --host_javabase=@bazel_tools//tools/jdk:absolute_javabase \
+        --define=ABSOLUTE_JAVABASE=$JAVA_HOME \
+        --host_java_toolchain=@bazel_tools//tools/jdk:toolchain_vanilla \
+        --java_toolchain=@bazel_tools//tools/jdk:toolchain_vanilla \
```

Comment:
What do these lines do? It looks similar to what https://github.com/NixOS/nixpkgs/pull/51783/files#diff-4746b81564a01e241a84d455575437e7R191 does in one line.

Reply:
`--server_javabase` set the java version used for the Bazel server, `--host_javabase` sets the java version used for the worker processes, `--*java_toolchain` sets the toolchain used to build java code.

I'm surprised setting `--server_javabase` is enough, maybe the other two default to the value passed for that.

Why I pass it separately here is that in the `checkPhase` we haven't wrapped the program yet (#51783 deletes `checkPhase` so it doesn't have that issue).